src/_guides/libraries/objective-c-interop.md

```diff
@@ -293,6 +293,60 @@ then check the status, and wait for the duration of the audio file:
     }
 ```
 
+### Callbacks and multithreading limitations
+
+Multithreading issues are the biggest limitation
+of Dart's experimental support for Objective-C interop.
+These limitations are due to the relationship between
+Dart isolates and OS threads,
+and the way Apple's APIs handle multithreading:
+
+* Dart isolates are not the same thing as threads.
+  Isolates run on threads,
+  but aren't guaranteed to run on any particular thread,
+  and the VM might change which thread an isolate is running on
+  without warning.
+  There is an [open feature request][] to allow isolates to be
+  pinned to specific threads.
+* While `ffigen` supports converting
+  Dart functions to Objective-C blocks,
+  most Apple APIs don't make any guarantees about
+  on which thread a callback will run.
+* Most APIs that involve UI interaction
+  can only be called on the main thread,
+  also called the _platform_ thread in Flutter.
+* Many Apple APIs are [not thread safe][].
+
+The first two points mean that a callback created in one isolate
+might be invoked on a thread running a different isolate,
```

Comment:
```suggestion
might be invoked on a thread running on a different isolate,
```

Reply:
I'm not sure about this one, as the isolates run on threads, not the other way around.